### ISSUE TITLE
Standardize and clean up dapr child helm chart values and behaviors.

### ISSUE THREAD

**Issue by CodeMonkeyLeet** [dapr/dapr#3257]
## In what area(s)?

/area helm

## Describe the proposal

Using the [advanced instructions to deploy Dapr using Helm](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-deploy/#install-with-helm-advanced), it feels that there are a range of inconsistencies in the way some [global values](https://github.com/dapr/dapr/blob/master/charts/dapr/README.md) interact with ones defined by child charts in a dapr deployment, particularly with the dashboard service. 

This proposal discusses standardizing how some common settings are handled and cleaning up the helm charts in general.
For example, the helm charts also fail `helm lint` today and should probably be fixed up so that they can pass that minimum bar.

| Value | Behavior | Proposal | Breaking Change? |
| --- | --- | --- | --- |
| `dapr_*.nameOverride`, `dapr_*.fullnameOverride` | These are defined in all the charts, but are undocumented and do not appear to be used. They are referenced in the `_helper.tpl` templates, but no other yaml/tpl files reference the definitions where the values are used. | Remove these definitions and `_helper.tpl` | Yes, but should have no functional impact. |
| `global.registry`, `global.tag` | Dashboard does not respect the global settings and only use the `dapr_dashboard.image.registry` and `tag` values, unlike the other charts. Dashboard is also unique in that it separates the custom `image.registry` from the `image.name` value where the other charts only provide the `image.name` value that must be provided as a _registry/name_ combined string, which is unintuitive. | Separate custom image registry and name values for all charts. Add custom tag value for all charts. Standardize all charts to default to global registry and tag values by default. | Yes. |
| `global.imagePullSecrets` | Dashboard ignores the global value in favor of its locally defined `dapr_dashboard.imagePullSecrets`, which no other chart defines. | Standardize all charts to default to global value. Add `imagePullSecrets` to all charts so that it can be customized on a per chart basis, since it facilitates existing support for each chart pulling from a custom registry. | Not API breaking but Dashboard values have changed semantics. |
| `global.ha.enabled`, `global.ha.replicaCount` | Dashboard checks `global.ha.enabled`, but ignores the `global.ha.replicaCount` and always applies its own `dapr_dashboard.replicaCount`. | Standardize to custom replicaCount only being applied if `global.ha.enabled` is true, and respect `global.ha.replicaCount` if custom `replicaCount` is not provided. | Not API breaking but Dashboard values have changed semantics. |
| `global.prometheus.enabled`, `global.prometheus.port` | Dashboard does not support these values. | Add annotations and ports for prometheus to dashboard deployment template so that is can also be scraped by prometheus. | No. |
| `global.logAsJson` | Not supported by Dashboard chart, though the dashboard binary does not support the `--log-as-json` arg anyway | Discrepancy in dashboard implementation, but not an issue specific to helm charts. See next item though. | N/A |
| `dapr_dashboard.logLevel` | This is specified for Dashboard, but it is not used, and like `--log-as-json` does not appear to be supported by the dashboard binary. | Remove the `dapr_dashboard.logLevel` value. | Yes, but should have no functional impact. |
| `dapr_dashboard.serviceType` | The Kubernetes service type config option is only available for customization for Dashboard, which makes sense if it is the only one that should have options to expose as an external port. | None unless we decide that this should be added to all the other charts as well. | N/A |
| `dapr_*.debug.enabled`, `dapr_*.debug.port`, `dapr_*.debug.initialDelaySeconds` | Dashboard does not support any of the debug settings unlike the other charts. Only the `debug.enabled` option is documented for the charts that support it. | Add debug options support to Dashboard chart. Document all debug option values. As a convenience option, define global versions of the settings since they apply to all charts. | No. |
| `dapr_*.runAsNonRoot` | All charts define and use this, but there's no global setting for this. | As a convenience option, define a global version of the setting. | No. |

**Comment by daixiang0** [comment]
I would pick `dapr.area.xxx` format.

**Comment by CodeMonkeyLeet** [comment]
> I would pick `dapr.area.xxx` format.

Are you suggesting converting the values of the format `dapr_dashboard` to the format `dapr.area.dashboard`? I'm a little hesitant to recommend that because these values are actually expressed as a naming hierarchy in yaml, so a user would have to nest their declarations for no apparent benefit, because there are no values that would be defined under the `dapr` or `dapr.area` tiers.

```yaml
dapr:
  area:
    dashboard:
      runAsNonRoot: "true"
```

My understanding is also that `global` is a special namespace in Helm, and would not be interchangeable with `dapr`.

Naming hierarchy aside, this would also be a breaking change for all charts, which would probably be nice to avoid if the only reasons for it are aesthetic?

**Comment by daixiang0** [comment]
No, I mean it like `dapr.dashboard.xxx`, `dashboard` is the area.

**Comment by CodeMonkeyLeet** [comment]
I see, that makes more sense. Even so, the guidance for helm charts is to [prefer flat values](https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values). 

I think changing the naming scheme for the dapr values is an interesting discussion, but one that probably deserves its own issue. There are a couple of other changes that are probably useful, such as converting from snake_cased to camelCased values names (e.g. `daprDashboard`) per best-practices etc.

**Comment by CodeMonkeyLeet** [comment]
In a related, separate feature request https://github.com/dapr/dapr/issues/3534, there is an ask to support `global.tolerations` or similar mechanism for more refined node targeting in addition to the existing `global.nodeSelector`.
